full-moon/src/tokenizer.rs

```diff
@@ -1,83 +1,327 @@
 use crate::{
+    atom::{trim_bracket_head, Atom},
     visitors::{Visit, VisitMut, Visitor, VisitorMut},
     ShortString,
 };
 
-use full_moon_derive::symbols;
+use logos::{Lexer, Logos, Span};
 #[cfg(feature = "serde")]
 use serde::{Deserialize, Serialize};
-use std::{cmp::Ordering, fmt, str::FromStr};
-
-symbols!(
-    And => "and",
-    Break => "break",
-    Do => "do",
-    ElseIf => "elseif",
-    Else => "else",
-    End => "end",
-    False => "false",
-    For => "for",
-    Function => "function",
-    If => "if",
-    In => "in",
-    Local => "local",
-    Nil => "nil",
-    Not => "not",
-    Or => "or",
-    Repeat => "repeat",
-    Return => "return",
-    Then => "then",
-    True => "true",
-    Until => "until",
-    While => "while",
+use std::{
+    cmp::Ordering,
+    convert::{TryFrom, TryInto},
+    fmt::{self, Display},
+};
+
+#[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
+#[cfg_attr(feature = "serde", serde(rename_all = "lowercase"))]
+#[allow(missing_docs)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum Symbol {
```

Comment:
Missing `non_exhaustive` attribute: https://github.com/Kampfkarren/full-moon/blob/main/full-moon-derive/src/symbols.rs#L98

Reply:
Also missing the doc comment above it too: `/// A literal symbol, used for both words important to syntax (like while) and operators (like +)`